app/monitoring/update-delivery-from-response.js

```diff
@@ -3,17 +3,29 @@ const { DELIVERED, PERMANENT_FAILURE, TEMPORARY_FAILURE, TECHNICAL_FAILURE } = r
 const completeDelivery = require('./complete-delivery')
 const failed = require('./failed')
 const rescheduleDelivery = require('./reschedule-delivery')
+const scheduleLetter = require('./schedule-letter')
+const { EMAIL } = require('../constants/methods')
 
 const updateDeliveryFromResponse = async (delivery, response) => {
   switch (response.data?.status) {
     case DELIVERED:
       await completeDelivery(delivery.deliveryId)
       break
     case PERMANENT_FAILURE:
-      await failed(delivery, INVALID)
+      await failed(delivery, {
+        reason: INVALID
+      })
+      if (delivery.method === EMAIL) {
+        await scheduleLetter(delivery)
+      }
       break
     case TEMPORARY_FAILURE:
-      await failed(delivery, REJECTED)
+      await failed(delivery, {
+        reason: REJECTED
+      })
+      if (delivery.method === EMAIL) {
+        await scheduleLetter(delivery)
+      }
```

Comment:
I think it is correct because the description in notify says:

Reply:
The provider could not deliver the message. This can happen when the recipient’s inbox is full or their anti-spam filter rejects your email. [Check your content does not look like spam](https://www.gov.uk/service-manual/design/sending-emails-and-text-messages#protect-your-users-from-spam-and-phishing) before you try to send the message again.

Reply:
which suggests sending another email isn't going to resolve it... p.s. sorry I pressed enter too many times and so made this into a few comments!